studio/tests/pages/projects/logs-query.test.js

```diff
@@ -139,17 +141,52 @@ test('custom sql querying', async () => {
 
 test('datepicker interaction updates query params', async () => {
   render(<LogsExplorerPage />)
-  clickDropdown(await screen.findByText(/Last day/))
+  clickDropdown(await screen.findByText(/Last 24 hours/))
   userEvent.click(await screen.findByText(/Last 3 days/))
 
-  const router = useRouter()
-  expect(router.push).toBeCalledWith(
-    expect.objectContaining({
-      query: expect.objectContaining({
-        its: expect.any(String),
-      }),
-    })
-  )
+  useProjectSubscription.mockReturnValue({
+    subscription: {
+      tier: {
+        supabase_prod_id: 'tier_pro',
+      },
+    },
+  })
+
+  const { subscription } = useProjectSubscription()
+
+  if (subscription.tier.supabase_prod_id === 'tier_pro') {
+    const router = useRouter()
+    expect(router.push).toBeCalled()
+    expect(router.push).toBeCalledWith(
+      expect.objectContaining({
+        query: expect.objectContaining({
+          its: expect.any(String),
+        }),
+      })
+    )
+  }
+})
+
+test.only('Upgrade prompt displays if user on free plan and tries to view beyond 24hrs', async () => {
+  render(<LogsExplorerPage />)
+  render(<UpgradePrompt />)
+
+  clickDropdown(await screen.findByText(/Last 24 hours/))
+  userEvent.click(await screen.findByText(/Last 3 days/))
+
+  useProjectSubscription.mockReturnValue({
+    subscription: {
+      tier: {
+        supabase_prod_id: 'free',
+      },
+    },
+  })
+
+  const { subscription } = useProjectSubscription()
```

Comment:
why would we need to get the mocked value when we already know that it is a free user? 
The mock is configured to [free by default](https://github.com/supabase/supabase/blob/9b9336632fbe32f0ce48c438c504dc9af14f6635/studio/__mocks__/hooks.js#L3)

Reply:
see also above comment